### DISCUSSION TITLE
moving named workspaces to another monitor

### DISCUSSION THREAD

**Discussion by d3m1gd** [YaLTeR/niri#712] [Ideas]
currently  if monitor is unplugged, all named workspaces are moved to the one still present, but upon connecting of another monitor, there is no way to move some of the named workspaces to that new monitor.

in general, for multimonitor setup it would be awesome to implement named workspaces affinity, where they can jump to any of the indicated monitors in config.

right now i have in config 
```
workspace "w4" { open-on-output "HDMI-A-1"; } 
```
but despite second monitor being connected is also shown as HDMI-A-1 (despite being physically different), no workspaces flow onto it, leaving it with single unnamed workspace.

**Comment by YaLTeR** [comment]
Hey, this option only sets the initial output that the workspace is created ("opens") on. Subsequently, all workspaces use common logic to move between outputs. You can move them manually with `move-window-to-monitor-left` etc. actions/binds.

In general, workspaces will remember the monitor they were on (by model/make/serial, not by connector) and return there (unless you open new windows on a workspace, then niri resets that workspace's original output to the current one).

**Reply by d3m1gd** [reply]
`move-window-to-monitor-left` etc are not exactly fitting for a script to automatucally move workspaces to a certain monitor,  as monitor position can change in config,  adding sometging like `move-to-monitor --output xxx` would solve that

**Reply by YaLTeR** [reply]
Yeah we should add that I guess
